src/MongoDB/BSON/functions.cpp

```diff
@@ -99,6 +99,12 @@ Variant HHVM_FUNCTION(MongoDBBsonToJson, const String &data)
 
 		str = bson_as_json(b, &str_len);
 
+		if (!str) {
+			throw MongoDriver::Utils::throwUnexpectedValueException("Could not convert BSON document to JSON");
+			bson_reader_destroy(reader);
```

Comment:
That doesn't work. The throw is a C++ exception which immediately returns from the function. The destroy should be first, and the return Variant() can be removed.